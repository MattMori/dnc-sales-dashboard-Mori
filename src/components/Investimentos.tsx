import React from "react";
import { Card, CardContent, Typography, Grid, List, ListItem } from "@mui/material";

interface Investimento {
    _id: string;
    nomeAtivo: string;
    tipoInvestimento: string;
    valorInvestido: number;
    plataforma: string;
}

interface InvestimentosProps {
    investimentos: Investimento[];
}

const Investimentos: React.FC<InvestimentosProps> = ({ investimentos }) => {
    return (
        <Card variant="outlined" sx={{ mb: 3 }}>
            <CardContent>
                <Typography variant="h6">Investimentos</Typography>
                <List>
                    {investimentos.map((investimento) => (
                        <ListItem key={investimento._id}>
                            <Grid container spacing={2}>
                                <Grid item xs={3}>
                                    <Typography>{investimento.nomeAtivo}</Typography>
                                </Grid>
                                <Grid item xs={3}>
                                    <Typography>{investimento.tipoInvestimento}</Typography>
                                </Grid>
                                <Grid item xs={3}>
                                    <Typography>R$ {investimento.valorInvestido}</Typography>
                                </Grid>
                                <Grid item xs={3}>
                                    <Typography>{investimento.plataforma}</Typography>
                                </Grid>
                            </Grid>
                        </ListItem>
                    ))}
                </List>
            </CardContent>
        </Card>
    );
};

export default Investimentos;
