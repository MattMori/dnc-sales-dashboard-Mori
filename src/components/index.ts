export { default as CustomTable} from './CustomTable';
export { default as FormComponent } from './Form-component';
export { default as Header } from './Header';
export {default as AvatarList} from './AvatarList';
export {default as CustomChart} from './Customchart';
export * from './BannerImage';
export * from './cardComponent';
export * from './Logo'
export * from './StyledButton';
export * from './StyledInput';
export * from './Typographies'