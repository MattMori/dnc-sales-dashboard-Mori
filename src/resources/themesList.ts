import { darkTheme, lightTheme } from '@/styles'

export const themesList = [
  { name: 'LightTheme', theme: lightTheme },
  { name: 'DarkTheme', theme: darkTheme },
]
