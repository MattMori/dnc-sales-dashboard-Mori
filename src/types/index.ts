export * from './theme';
export * from './form-component';
export * from './typographies'
export * from './avatarList'