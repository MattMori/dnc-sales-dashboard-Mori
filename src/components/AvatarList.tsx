import { StyledH2, StyledSpan } from './Typographies'
import { Avatar, Box } from '@mui/material'
import { pxToRem } from '@/utils'
import { AvatarListProps } from '@/types'

function AvatarList(props: AvatarListProps) {
  return (
    <>
      {props.listData.map((item, index) => (
        <Box
          key={index}
          sx={{
            alignItems: 'center',
            display: 'flex',
            padding: `${pxToRem(10)} 0`,
          }}
        >
          <Box>
            <Avatar
              alt={item.name}
              src={item.avatar}
              sx={{
                height: pxToRem(48),
                width: pxToRem(48),
                marginRight: pxToRem(16),
              }}
            />
          </Box>
          <Box>
            <StyledH2>{item.name}</StyledH2>
            <StyledSpan>{item.subtitle}</StyledSpan>
          </Box>
        </Box>
      ))}
    </>
  )
}

export default AvatarList
