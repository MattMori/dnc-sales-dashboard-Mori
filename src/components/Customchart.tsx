import { useTheme } from "styled-components";
import {
    Chart as ChartJS,
    CategoryScale,
    LinearScale,
    BarElement,
    LineElement,
    PointElement,
    Title,
    Tooltip,
    Legend
} from 'chart.js';
import { Line } from 'react-chartjs-2';
import { Bar } from 'react-chartjs-2';
import { CustomChartProps } from '@/types';

ChartJS.register(
    CategoryScale,
    LinearScale,
    BarElement,
    LineElement,
    Title,
    Tooltip,
    Legend,
    PointElement,
);


function CustomChart(props: CustomChartProps) {
    const { data, labels, type } = props;
    const theme = useTheme();
    const options = {
        responsive: true,
        scaleShowVerticalLines: false,
        scales: {
            x: {
                border: {
                    display: false,
                },
                grid: {
                    display: false,
                },
                ticks: {
                    color: theme.typographies.subtitle,
                },
            },
            y: {
                border: {
                    display: false,
                },
                grid: {
                    color: theme.appDefaultStroke,
                },
                ticks: {
                    color: theme.typographies.subtitle,
                },
            },
        },
        plugins: {
            legend: {
                display: false,
            },
        },
    };

    const ChartData = {
        labels,
        datasets: [
            {
                data: data,
                borderColor: 'rgba(75, 192, 192, 1)',
                backgroundColor: 'rgb(11, 146, 146)',
            },
        ],
    }
    return type === 'bar' ? (<Bar options={options} data={ChartData} />) : (<Line options={options} data={ChartData} />);
}
export default CustomChart;
