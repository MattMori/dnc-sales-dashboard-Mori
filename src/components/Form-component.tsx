import styled from "styled-components";
import { formComponentProps } from "@/types";
import { StyledButton, StyledInput } from "@/components";
import { pxToRem } from "@/utils";

export const StyledForm = styled.form`
    display: flex;
    flex-direction: column;
    row-gap: ${pxToRem(16)};
`

function FormComponent(props: formComponentProps) {
    const { inputs, buttons, message } = props
    return (
        <StyledForm>
            {inputs.map((inputProps, index) => (
                <StyledInput key={index} {...inputProps} />
            ))}
            {buttons.map((buttonProps, index) => (
                <StyledButton key={index} {...buttonProps} />
            ))}
            {message && <p>{<div style={{ color: message.type === 'error' ? 'red' : 'green' }}>
                {message.msg}
            </div>}</p>}
        </StyledForm>
    )

}

export default FormComponent;