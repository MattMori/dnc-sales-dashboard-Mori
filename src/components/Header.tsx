import styled from 'styled-components'
import { Link } from 'react-router-dom'
import { Logo } from '@/components'
import { Box, Container, Typography } from '@mui/material'
import { pxToRem } from '@/utils'
import { useProfileImage } from '@/Hooks/useProfileImage'
import { useGet } from '@/Hooks'
import { ProfileData } from '@/types'
import { ProfileImage } from './ProfileImage'
import PersonIcon from '@mui/icons-material/Person'

const StyledHeader = styled.header`
  background-color: ${(props) => props.theme.appBackground};
  border-bottom: ${pxToRem(1)} solid ${(props) => props.theme.appDefaultStroke};
  margin-bottom: ${pxToRem(37)};
  width: 100%;
`

export const Header = () => {
  const [profileImage] = useProfileImage()
  const { data: profileData } = useGet<ProfileData>('/usuario/dados')
  const userName = profileData?.resposta?.name || ''

  return (
    <StyledHeader>
      <Container maxWidth="lg">
        <Box
          sx={{
            display: 'flex',
            alignItems: 'center',
            justifyContent: 'space-between',
            height: pxToRem(64),
          }}
        >
          <Link to="/home">
            <Logo height={30} width={73} />
          </Link>
          <Box
            sx={{
              display: 'flex',
              alignItems: 'center',
              gap: 2,
            }}
          >
            <Link to="/perfil">
              <Box
                sx={{
                  width: 40,
                  height: 40,
                  borderRadius: '50%',
                  overflow: 'hidden',
                  display: 'flex',
                  alignItems: 'center',
                  justifyContent: 'center',
                  border: '1px solid #ccc',
                  '&:hover': {
                    opacity: 0.8,
                  },
                }}
              >
                {profileImage ? (
                  <ProfileImage initialImage={profileImage} size="small" />
                ) : (
                  <PersonIcon sx={{ fontSize: 24, color: '#ccc' }} />
                )}
              </Box>
            </Link>
            <Typography
              variant="subtitle1"
              sx={{
                color: 'text.primaryColor',
                fontWeight: 500,
              }}
            >
              {userName}
            </Typography>
          </Box>
        </Box>
      </Container>
    </StyledHeader>
  )
}

export default Header
