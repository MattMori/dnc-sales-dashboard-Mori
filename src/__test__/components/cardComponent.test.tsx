import { CardComponent } from '@/components/cardComponent'
import { render } from '@testing-library/react'
import { Theme } from '@/types/theme'
import { ThemeProvider } from 'styled-components'
import { themesList } from '@/resources/themesList'

describe('cardComponent', () => {
  const renderComponent = (theme: Theme, className?: string) =>
    render(
      <ThemeProvider theme={theme}>
        <CardComponent className={className} />
      </ThemeProvider>,
    )

  themesList.forEach(({ name, theme }) => {
    describe(`${name}`, () => {
      it('should math the snapshot without any class', () => {
        const { asFragment } = renderComponent(theme)
        expect(asFragment()).toMatchSnapshot()
      })

      it('should math the snapshot with alert class', () => {
        const { asFragment } = renderComponent(theme, 'alert')
        expect(asFragment()).toMatchSnapshot()
      })

      it('should math the snapshot with success class', () => {
        const { asFragment } = renderComponent(theme, 'success')
        expect(asFragment()).toMatchSnapshot()
      })

      it('should math the snapshot with warning class', () => {
        const { asFragment } = renderComponent(theme, 'warning')
        expect(asFragment()).toMatchSnapshot()
      })
    })
  })
})
