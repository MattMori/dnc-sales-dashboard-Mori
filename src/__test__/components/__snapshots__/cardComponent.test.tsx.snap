// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cardComponent DarkTheme should math the snapshot with alert class 1`] = `
<DocumentFragment>
  .c0 {
  background-color: #0F1535;
  border: 0.0625rem solid #21497D;
  border-radius: 0.5rem;
  box-sizing: border-box;
  padding: 1.5rem;
  width: 100%;
}

.c0.alert {
  background-color: #E80000;
  border-color: #E80000;
}

.c0.success {
  background-color: #008000;
  border-color: #008000;
}

.c0.warning {
  background-color: #F7A300;
  border-color: #F7A300;
}

.c0 a {
  color: inherit;
  text-decoration: transparent;
}

<div
    class="c0 alert"
  />
</DocumentFragment>
`;

exports[`cardComponent DarkTheme should math the snapshot with success class 1`] = `
<DocumentFragment>
  .c0 {
  background-color: #0F1535;
  border: 0.0625rem solid #21497D;
  border-radius: 0.5rem;
  box-sizing: border-box;
  padding: 1.5rem;
  width: 100%;
}

.c0.alert {
  background-color: #E80000;
  border-color: #E80000;
}

.c0.success {
  background-color: #008000;
  border-color: #008000;
}

.c0.warning {
  background-color: #F7A300;
  border-color: #F7A300;
}

.c0 a {
  color: inherit;
  text-decoration: transparent;
}

<div
    class="c0 success"
  />
</DocumentFragment>
`;

exports[`cardComponent DarkTheme should math the snapshot with warning class 1`] = `
<DocumentFragment>
  .c0 {
  background-color: #0F1535;
  border: 0.0625rem solid #21497D;
  border-radius: 0.5rem;
  box-sizing: border-box;
  padding: 1.5rem;
  width: 100%;
}

.c0.alert {
  background-color: #E80000;
  border-color: #E80000;
}

.c0.success {
  background-color: #008000;
  border-color: #008000;
}

.c0.warning {
  background-color: #F7A300;
  border-color: #F7A300;
}

.c0 a {
  color: inherit;
  text-decoration: transparent;
}

<div
    class="c0 warning"
  />
</DocumentFragment>
`;

exports[`cardComponent DarkTheme should math the snapshot without any class 1`] = `
<DocumentFragment>
  .c0 {
  background-color: #0F1535;
  border: 0.0625rem solid #21497D;
  border-radius: 0.5rem;
  box-sizing: border-box;
  padding: 1.5rem;
  width: 100%;
}

.c0.alert {
  background-color: #E80000;
  border-color: #E80000;
}

.c0.success {
  background-color: #008000;
  border-color: #008000;
}

.c0.warning {
  background-color: #F7A300;
  border-color: #F7A300;
}

.c0 a {
  color: inherit;
  text-decoration: transparent;
}

<div
    class="c0"
  />
</DocumentFragment>
`;

exports[`cardComponent LightTheme should math the snapshot with alert class 1`] = `
<DocumentFragment>
  .c0 {
  background-color: #FFF;
  border: 0.0625rem solid #E0E0E0;
  border-radius: 0.5rem;
  box-sizing: border-box;
  padding: 1.5rem;
  width: 100%;
}

.c0.alert {
  background-color: #E80000;
  border-color: #E80000;
}

.c0.success {
  background-color: #008000;
  border-color: #008000;
}

.c0.warning {
  background-color: #F7A300;
  border-color: #F7A300;
}

.c0 a {
  color: inherit;
  text-decoration: transparent;
}

<div
    class="c0 alert"
  />
</DocumentFragment>
`;

exports[`cardComponent LightTheme should math the snapshot with success class 1`] = `
<DocumentFragment>
  .c0 {
  background-color: #FFF;
  border: 0.0625rem solid #E0E0E0;
  border-radius: 0.5rem;
  box-sizing: border-box;
  padding: 1.5rem;
  width: 100%;
}

.c0.alert {
  background-color: #E80000;
  border-color: #E80000;
}

.c0.success {
  background-color: #008000;
  border-color: #008000;
}

.c0.warning {
  background-color: #F7A300;
  border-color: #F7A300;
}

.c0 a {
  color: inherit;
  text-decoration: transparent;
}

<div
    class="c0 success"
  />
</DocumentFragment>
`;

exports[`cardComponent LightTheme should math the snapshot with warning class 1`] = `
<DocumentFragment>
  .c0 {
  background-color: #FFF;
  border: 0.0625rem solid #E0E0E0;
  border-radius: 0.5rem;
  box-sizing: border-box;
  padding: 1.5rem;
  width: 100%;
}

.c0.alert {
  background-color: #E80000;
  border-color: #E80000;
}

.c0.success {
  background-color: #008000;
  border-color: #008000;
}

.c0.warning {
  background-color: #F7A300;
  border-color: #F7A300;
}

.c0 a {
  color: inherit;
  text-decoration: transparent;
}

<div
    class="c0 warning"
  />
</DocumentFragment>
`;

exports[`cardComponent LightTheme should math the snapshot without any class 1`] = `
<DocumentFragment>
  .c0 {
  background-color: #FFF;
  border: 0.0625rem solid #E0E0E0;
  border-radius: 0.5rem;
  box-sizing: border-box;
  padding: 1.5rem;
  width: 100%;
}

.c0.alert {
  background-color: #E80000;
  border-color: #E80000;
}

.c0.success {
  background-color: #008000;
  border-color: #008000;
}

.c0.warning {
  background-color: #F7A300;
  border-color: #F7A300;
}

.c0 a {
  color: inherit;
  text-decoration: transparent;
}

<div
    class="c0"
  />
</DocumentFragment>
`;
